pulsar-broker/src/test/java/org/apache/pulsar/client/impl/ProducerCloseTest.java

```diff
@@ -73,6 +74,31 @@ public void testProducerCloseCallback() throws Exception {
         Assert.assertEquals(completableFuture.isDone(), true);
     }
 
+    @Test(timeOut = 10_000)
+    public void testProducerCloseFailsPendingBatchWhenPreviousStateNotReadyCallback() throws Exception {
+        initClient();
+        @Cleanup
+        ProducerImpl<byte[]> producer = (ProducerImpl<byte[]>) pulsarClient.newProducer()
+                .topic("testProducerClose")
+                .maxPendingMessages(10)
+                .batchingMaxPublishDelay(10, TimeUnit.SECONDS)
+                .batchingMaxBytes(Integer.MAX_VALUE)
+                .enableBatching(true)
+                .create();
+        final TypedMessageBuilder<byte[]> messageBuilder = producer.newMessage();
+        final TypedMessageBuilder<byte[]> value = messageBuilder.value("test-msg".getBytes(StandardCharsets.UTF_8));
+        final CompletableFuture<MessageId> completableFuture = value.sendAsync();
```

Comment:
This could be shortened to:
```suggestion
        CompletableFuture<MessageId> completableFuture = producer.newMessage()
            .value("test-msg".getBytes(StandardCharsets.UTF_8))
            .sendAsync();
```